### PR TITLE
Monitoring and telemetry document

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Closes: https://github.com/keep-network/pm/issues/33

Here we add a document describing the entire monitoring and telemetry system.

### CODE REVIEW

File: monitoring/docs/monitoring-and-telemetry.adoc [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+== Sources of data
+
+Each source produces a different kind of monitoring and telemetry data. Here is
+a detailed description of each data source.
+
+=== Threshold dashboard
+
+The dashboard instances collect the following telemetry data that are sent
+directly to Sentry hub:
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply monitoring/docs/monitoring-and-telemetry.adoc:43]: That just means the "instances run by users in their browsers". Nothing specific.

File: monitoring/docs/monitoring-and-telemetry.adoc [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    D[TBTCv2 minters and guardians] --> |Telemetry| C
+    C -->|Alerts| E(PagerDuty)
+    C --> |Alerts| F(Discord)
+    G[TBTCv2 monitoring] --> |Alerts| C
+    G --> |Notifications| F
+....
+
+There are several sources of monitoring and telemetry data:
+
+* https://github.com/threshold-network/token-dashboard[Threshold dashboard] + https://github.com/keep-network/v2-end-to-end[second track deposit verifier]
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply monitoring/docs/monitoring-and-telemetry.adoc:26]: We can do it separately.

File: monitoring/docs/monitoring-and-telemetry.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (pdyraga) [reply monitoring/docs/monitoring-and-telemetry.adoc:43]: What I mean is:

```suggestion
directly to Sentry hub:

* generated deposit data,
* unhandled errors.
```

File: monitoring/docs/monitoring-and-telemetry.adoc [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+==== Unhandled errors
+
+Apart from deposit data, all unhandled errors thrown by the dashboard are also
+captured by the telemetry system. This allows to identify bugs encountered by
+users in an automatic way.
+
+=== TBTCv2 minters and guardians
+
+The minters and guardians instances collect the following telemetry data that
+are sent directly to Sentry hub:
```
Comment (pdyraga) [inline monitoring/docs/monitoring-and-telemetry.adoc:80]:
```suggestion
are sent directly to Sentry hub:

* handled processing and validation errors,
* unhandled errors.
```

File: monitoring/docs/monitoring-and-telemetry.adoc [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+=== TBTCv2 monitoring
+
+The monitoring component inspects the tBTC v2 system on-chain contracts and
+produce different kind of system events that are sent to Sentry and Discord
+based on their type. A general rule of thumb is that notifications
+(*i.e. informational sytem events*) are sent directly to Discord `pass:[*]-notifications`
+channels while alerts requiring an action (*i.e. warning/critical system events*)
+are propagated to the Sentry hub that decides about next steps. Specific system
+events produced by the monitoring component are:
```
Comment (pdyraga) [inline monitoring/docs/monitoring-and-telemetry.adoc:105]:
```suggestion
events produced by the monitoring component are:

* deposit revealed,
* wallet registered,
* large deposit revealed,
* optimistic minting canceled,
* optimistic minting requested too early,
* optimistic minting requested for undetermined Bitcoin transaction.
```

It is not a huge problem but I think this version is easier to understand by someone who reads about the monitoring for the first time.

File: monitoring/docs/monitoring-and-telemetry.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply monitoring/docs/monitoring-and-telemetry.adoc:43]: Ah ok! I was sure you are talking about specific dashboard instances 🤦 Here it is: e769513f

File: monitoring/docs/monitoring-and-telemetry.adoc (hunk 3 above)
Comment (lukasz-zimnoch) [reply monitoring/docs/monitoring-and-telemetry.adoc:83]: See e769513f

File: monitoring/docs/monitoring-and-telemetry.adoc (hunk 4 above)
Comment (lukasz-zimnoch) [reply monitoring/docs/monitoring-and-telemetry.adoc:111]: See e769513f

**Review by pdyraga — APPROVED** [review]

**Pull request merged**